packages/grafana-ui/src/components/Table/TableNG/TableNG.tsx

```diff
@@ -502,6 +505,12 @@ export function TableNG(props: TableNGProps) {
             const cellColorStyles = getCellColorInlineStyles(cellOptions, displayValue, applyToRowBgFn != null);
             Object.assign(style, cellColorStyles);
           }
+          if (styleField) {
+            const cellStyleByField = parseStyleJson(props.row[getDisplayName(styleField)]);
+            if (cellStyleByField) {
+              style = { ...style, ...cellStyleByField };
+            }
+          }
```

Comment:
addl perf notes:

- `if` is always fastest with a boolean that doesnt need type coercion. instead of `if (styleField)`, i'd extract this to `if (hasStyleField)`, similar to `canBeColorized`.
- `getDisplayName(styleField)` can be pulled up into a const next to the _other_ `styleFieldName` :sweat_smile: 
- the inner `if` is not needed at all, you can just spread the undefined. this will likely be an extremely rare case as well, so we dont need to eat the cost of the extra `if` check.

Reply:
Good call on all of this. Looked at it more this morning and decided that the boolean we store ought to be `hasValidStyleField`, which is both that a `styleField` is set, and that we _found_ the field when we searched the fields by that name. We *could* eventually expand the definition of "valid" if that's not enough, like for example, we could only allow that field to be of type `string`, although that just adds more processing time in the default case, and `parseStyleJson` just punts if you mess that up, so to me it doesn't seem worth adding anything else here yet, but lmk if you disagree.